apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -123,6 +126,9 @@ const aiTutorSlice = createSlice({
     addAIResponse: (state, action: PayloadAction<string | undefined>) => {
       state.aiResponse = action.payload;
     },
+    setLevel: (state, action: PayloadAction<Level | undefined>) => {
+      state.level = action.payload;
+    },
```

Comment:
❓ Is there a way to call `setLevel` with the aitutorleveldata in `levels/show.js` rather than passing it down to all of the children and grandchildren components? Basically something like [this](https://github.com/code-dot-org/code-dot-org/blob/75443933100d3d3e7bd45da546588379c7d18cc7/apps/src/sites/studio/pages/levels/show.js#L84) but I wasn't sure what syntax to use with TypeScript and the way `aiTutorRedux` is set up.

Reply:
I think we generally have a preference to avoid using redux code outside of React files (even though we have plenty of exceptions in older code 😅). I'm not as familiar with this feature, but where is the level information originating from? Is this tool meant to be used with any lab, or just Java Lab?